platform/linux-generic/pktio/socket.c

```diff
@@ -274,9 +274,16 @@ static int sock_mmsg_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 			if (msgvec[i].msg_hdr.msg_iov->iov_len < pkt_len)
 				seg_len = msgvec[i].msg_hdr.msg_iov->iov_len;
 
-			if (_odp_cls_classify_packet(pktio_entry, base, pkt_len,
-						     seg_len, &pool, pkt_hdr,
-						     true)) {
+			packet_set_len(pkt_hdr, pkt_len);
```

Comment:
It seems that odp_packet_trunc_tail() should be done instead of packet_set_len(), otherwise, e.g. tailroom will not be updated. So, trunc_tail() call (from couple of line below) could be moved up to before pktio_cls_enabled() line. It would need to be called anyway.